src/main/java/stsjorbsmod/cards/wanderer/ChainLightning.java

```diff
@@ -27,16 +32,54 @@
     private static final int DAMAGE_PLUS_PER_HOP = 2;
     private static final int UPGRADE_PLUS_PER_HOP = 2;
 
+    private int damageMultiplier = 0;
+
     public ChainLightning() {
         super(ID, IMG, COST, TYPE, COLOR, RARITY, TARGET);
         damage = baseDamage = DAMAGE;
         magicNumber = baseMagicNumber = DAMAGE_PLUS_PER_HOP;
     }
 
+    @Override
+    protected int calculateBonusBaseDamage() {
+        return magicNumber * damageMultiplier;
+    }
+
     @Override
     public void use(AbstractPlayer p, AbstractMonster m) {
-        ArrayList<AbstractMonster> targets = AbstractDungeon.getMonsters().monsters;
-        addToBot(new ChainLightningAction(p, m, targets, damage, magicNumber, AttackEffect.NONE));
+        ArrayList<AbstractMonster> targets = getRandomOrderMonsters(AbstractDungeon.getMonsters().monsters, m);
+        this.damageMultiplier = 0;
+        for (AbstractMonster monster: targets) {
+            this.calculateCardDamage(monster);
+            AbstractDungeon.actionManager.addToBottom(new DamageAction(
+                    monster,
+                    new DamageInfo(p, this.damage, DamageInfo.DamageType.NORMAL),
+                    AttackEffect.NONE));
+            addLightningEffect(monster, this.damageMultiplier);
+            this.damageMultiplier += 1;
+        }

```

Comment:
This refactor looks much better! I think the "reseting `damageMultiplier` at the end of use" that was here previously needs to be put back, though, otherwise I think the highest `damageMultiplier` from the last time it was played will get calculated into the damage displayed on the card description the next time through the deck

Reply:
Good point! I’ll add that back tomorrow and we can get this merged.

Reply:
I'll go ahead and add it so I can include it in a release tonight. Thanks for the work!